src/RequestProcessor.cs

```diff
@@ -209,6 +211,28 @@ internal StreamingMessage ProcessInvocationRequest(StreamingMessage request)
                 functionInfo = _functionLoader.GetFunctionInfo(request.InvocationRequest.FunctionId);
                 psManager = _powershellPool.CheckoutIdleWorker(request, functionInfo);
 
+                if (_dependencyDownloadTask != null
+                    && ((_dependencyDownloadTask.Status != TaskStatus.Canceled)
+                    || _dependencyDownloadTask.Status != TaskStatus.Faulted
+                    || _dependencyDownloadTask.Status != TaskStatus.RanToCompletion))
+                {
+                    psManager.Logger.Log(LogLevel.Information, PowerShellWorkerStrings.DependencyDownloadInProgress, null, true);
+                    psManager.Logger.Log(LogLevel.Information, PowerShellWorkerStrings.DependencyDownloadInProgress, null);
+                    _dependencyDownloadTask.Wait();
+                }
+
+                if (_dependencyManager?.DependencyError != null)
+                {
+                    StreamingMessage response = NewStreamingMessageTemplate(request.RequestId,
+                        StreamingMessage.ContentOneofCase.InvocationResponse,
+                        out StatusResult status);
+                    status.Status = StatusResult.Types.Status.Failure;
+                    status.Exception = _dependencyManager.DependencyError.ToRpcException();
+                    response.InvocationResponse.InvocationId = request.InvocationRequest.InvocationId;
+                    return response;
```

Comment:
As far as I understand, returning a error response to host in `FunctionLoadRequest` phase meaning a hard failure -- all subsequent invocation will fail with the same error.
Failure in `InvocationRequest` phase is not considered a hard failure. This means the next invocation comes in, and it will just continue to execute just like the dependency download/check was successful.

Reply:
Also, a similar issue -- we have to create a `Runspace` and evaluate `profile.ps1` in `FunctionLoad` phase, so that we get a hard failure if `profile.ps1` fails to run.

Reply:
Yes. FunctioLoad failure is preserved and is replayed on function invocation. Scope of an invocation request failure is only for that invocation

Reply:
If DependencyError download error occurs then for all invocation will receive this error and function execution wouldn't proceed

Reply:
You are basically moving the logic that's supposed to be handled by the host to the worker.

See my comment https://github.com/Azure/azure-functions-powershell-worker/pull/193#discussion_r277522973
I believe `profile.ps1` needs to be evaluated during function load phase, because a bad `profile.ps1` is a hard failure. But `profile.ps1` cannot be invoked until after the dependency download.

Reply:
Since dependency download happens on a separate instance of the PowerShell and doesn't use the runspace of powershell pool  hence I've added back the __powershellPool.Initialize(request.RequestId); as part of FunctionLoad